src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -794,8 +794,6 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
             // This method is written such that control generally flows top-to-bottom, avoiding
             // jumps as much as possible in the optimistic case of "all ASCII". If we see non-ASCII
             // data, we jump out of the hot paths to targets at the end of the method.
-
-            Vector128<ushort> asciiMaskForTestZ = Vector128.Create((ushort)0xFF80); // used for PTEST on supported hardware
             Vector128<ushort> asciiMaskForAddSaturate = Vector128.Create((ushort)0x7F80); // used for PADDUSW
```

Comment:
This is just used once in `FoundNonAsciiDataInFirstVector:`, but inside `VectorContainsNonAsciiChar` this is created. Can you reuse `asciiMaskForAddSaturate` inside `VectorContainsNonAsciiChar` instead? Not sure it get hoisted out of the loop.

Reply:
`VectorContainsNonAsciiChar` is inlined and,  both `asciiMaskForAddSaturate` and `asciiMaskForTestZ` are loaded from the [constants in the data section](https://gist.github.com/SwapnilGaikwad/5e40f3c7fe9cdd34425f0bfa67be37fc#file-x86_getindexoffirstnonasciichar_sse2_both_sse2_and_sse42-L222-L223). So it doesn't get reallocated and passing it as an arg would not change assembly.

Full asm for [GetIndexOfFirstNonAsciiChar_Sse2](https://gist.github.com/SwapnilGaikwad/5e40f3c7fe9cdd34425f0bfa67be37fc)